.github/workflows/github-actions-demo.yml

```diff
@@ -0,0 +1,26 @@
+name: leaderboard-backend CI
+
+on:
+  push:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+
+  Golang-CI:
+    runs-on: ubuntu-latest
```

Comment:
Ultra nit: I think this is the default it suggests, but I wonder if there is a lighter weight image that will save us on CI minutes.

Reply:
<img width="1083" alt="image" src="https://user-images.githubusercontent.com/9021608/125905530-7af0f88f-0ebe-4e80-92d2-d8e1fedddd11.png">
It seems just 10% of the runtime is spent on setting up the image, doesn't feel like it impacts our CI time much. Unless other images will perform better in other parts of the workflow as well?

Reply:
Also github actions are [free for open source](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#about-billing-for-github-actions):

> GitHub Actions usage is free for both public repositories and self-hosted runners. For private repositories, each GitHub account receives a certain amount of free minutes and storage, depending on the product used with the account. Any usage beyond the included amounts is controlled by spending limits.

Reply:
Of the actual [images](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources), there's nothing lighter weight without hosting our own. The list:
Windows Server 2019, Windows Server 2016
Ubuntu 20.04, Ubuntu 18.04
macOS Big Sur 11, macOS Catalina 10.15 

<!--EndFragment-->
</body>
</html>

Reply:
I think the time right now is fine, as _most_ of it is bootstrapping time that shouldn't grow significantly with the codebase. 1 minute is fine, we can optimize once it starts growing and gets to be noticeable for our workflow.